Sources/Account/Account+TransactionOperations.swift

```diff
@@ -56,7 +57,7 @@ extension Account {
             }
 
             let (unspentTxOuts, ledgerBlockCount) =
-                account.readSync { ($0.unspentTxOuts, $0.knowableBlockCount) }
+            account.readSync { ($0.unspentTxOuts(tokenId: amount.tokenId), $0.knowableBlockCount) }
```

Comment:
just an observation...

unspentTxOuts calls unspentTxOutsAndBlockCount (ref: Account.swift), which internally calls self.knowableBlockCount, which has to check `for txOut in allTxOutTrackers` to calculate the computed property value.

The separate call to get knowableBlockCount will re-do the calculation of the computed property for knowableBlockCount.

Not sure how impactful that calculation is, but since there's a call to get both from account in one call, maybe we should use that instead.

Reply:
think this is premature optimization as-well, bernie tried to optimize these loops and ran into some new bugs. Im happy to optimize in the future if its needed.

Reply:
verified not a bottleneck